app/views/internal/feedback_messages/_abuse_reports.html.erb

```diff
@@ -44,26 +44,61 @@
-                    <%= f.submit "CONFIRMED", class: "btn btn-success btn-sm", data: { confirm: "Are you sure?" } %>
-                  <% end %>
+                  <button class="btn btn-success btn-sm js-reaction-confirm" type="button" data-reactable="user" data-id="<%= reaction.id %>" data-status="confirmed">CONFIRM</button>
                 <% else %>
-                  <%= form_for [:internal, reaction], html: { class: "d-inline" } do |f| %>
-                    <%= f.hidden_field :status, value: "confirmed" %>
-                    <%= f.submit "CONFIRMED", class: "btn btn-success btn-sm" %>
-                  <% end %>
-                <% end %>
-                <%= form_for [:internal, reaction], html: { class: "d-inline" } do |f| %>
-                  <%= f.hidden_field :status, value: "invalid" %>
-                  <%= f.submit "INVALID", class: "btn btn-danger btn-sm" %>
+                  <button class="btn btn-success btn-sm js-reaction-confirm" type="button" data-reactable="not-user" data-id="<%= reaction.id %>" data-status="confirmed">CONFIRM</button>
                 <% end %>
+                <button class="btn btn-danger btn-sm js-reaction-confirm" type="button" data-id="<%= reaction.id %>" data-status="invalid">INVALID</button>
               <% end %>
             </span>
           </div>
-          <hr>
+          <hr id="js__reaction__div__hr__<%= reaction.id %>">
         <% end %>
       </div>
     </div>
   </div>
 </div>
+
+<script>
+  function updateReactionStatus(id, status){
+    fetch(`/internal/reactions/${id}`, {
+     method: 'PATCH',
+     headers: {
+      Accept: 'application/json',
+       'X-CSRF-Token': document.querySelector("meta[name='csrf-token']").content,
+       'Content-Type': 'application/json',
+     },
+     body: JSON.stringify({
+       id: id,
+       status: status,
+      }),
+     credentials: 'same-origin'
+   })
+     .then(response => response.json()
+                                     .then(json => {
+                                       if (json.outcome === 'Success') {
+                                        removeReactionDiv(id)
+                                       } else {
+                                         console.log(json.outcome);
+                                       }
+                                     })
+                                     .catch(error => {
+                                       console.log(error);
+                                     }))
+
+  }
+
+  function removeReactionDiv(id){
+    document.getElementById(`js__reaction__div__${id}`).remove();
+    document.getElementById(`js__reaction__div__hr__${id}`).remove();
+  }
+
+  document.querySelectorAll('.js-reaction-confirm').forEach(item => {
+    item.addEventListener('click', function () {
+      if (item.dataset.reactable && item.dataset.reactable === 'user') {
+        confirm("You are confirming a vomit reaction for a User; are you sure?");
```

Comment:
what happens if the user cancels and this returns false? it seems like `updateReactionStatus` runs no matter the prompt

Reply:
🤦‍♀ thanks for catching this